OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -34,103 +45,118 @@ public FindResources(CPos avoidCell)
+				mobileInfo = self.Info.Traits.Get<MobileInfo>();
+				pathFinder = self.World.WorldActor.Trait<IPathFinder>();
+				loadedVars = true;
+			}
 
 			if (harv.IsFull)
 				return Util.SequenceActivities(new DeliverResources(), NextActivity);
 
-			var harvInfo = self.Info.Traits.Get<HarvesterInfo>();
-			var mobile = self.Trait<Mobile>();
-			var mobileInfo = self.Info.Traits.Get<MobileInfo>();
-			var resLayer = self.World.WorldActor.Trait<ResourceLayer>();
-			var territory = self.World.WorldActor.TraitOrDefault<ResourceClaimLayer>();
+			var closestHarvestablePosition = ClosestHarvestablePos(self);
 
-			// Determine where to search from and how far to search:
-			var searchFromLoc = harv.LastOrderLocation ?? (harv.LastLinkedProc ?? harv.LinkedProc ?? self).Location;
-			var searchRadius = harv.LastOrderLocation.HasValue ? harvInfo.SearchFromOrderRadius : harvInfo.SearchFromProcRadius;
-			var searchRadiusSquared = searchRadius * searchRadius;
-
-			// Find harvestable resources nearby:
-			var path = self.World.WorldActor.Trait<IPathFinder>().FindPath(
-				PathSearch.Search(self.World, mobileInfo, self, true)
-					.WithHeuristic(loc =>
-					{
-						// Avoid this cell:
-						if (avoidCell.HasValue && loc == avoidCell.Value)
-							return EstimateDistance(loc, searchFromLoc) + Constants.CellCost;
-
-						// Don't harvest out of range:
-						var distSquared = (loc - searchFromLoc).LengthSquared;
-						if (distSquared > searchRadiusSquared)
-							return EstimateDistance(loc, searchFromLoc) + Constants.CellCost * 2;
-
-						// Get the resource at this location:
-						var resType = resLayer.GetResource(loc);
-						if (resType == null)
-							return EstimateDistance(loc, searchFromLoc) + Constants.CellCost;
-
-						// Can the harvester collect this kind of resource?
-						if (!harvInfo.Resources.Contains(resType.Info.Name))
-							return EstimateDistance(loc, searchFromLoc) + Constants.CellCost;
-
-						if (territory != null)
-						{
-							// Another harvester has claimed this resource:
-							ResourceClaim claim;
-							if (territory.IsClaimedByAnyoneElse(self, loc, out claim))
-								return EstimateDistance(loc, searchFromLoc) + Constants.CellCost;
-						}
-
-						return 0;
-					})
-					.FromPoint(self.Location));
-
-			if (path.Count == 0)
+			// If no harvestable position could be found, either deliver the remaining resources
+			// or get out of the way and do not disturb.
+			if (!closestHarvestablePosition.HasValue)
 			{
 				if (!harv.IsEmpty)
 					return new DeliverResources();
+
+				harv.UnblockRefinery(self);
+				var moveTo = harv.LastHarvestedCell ?? (self.Location + new CVec(0, 4));
+				self.QueueActivity(mobile.MoveTo(moveTo, 1));
+				self.SetTargetLine(Target.FromCell(self.World, moveTo), Color.Gray, false);
+
+				var randFrames = 125 + self.World.SharedRandom.Next(-35, 35);
+				if (NextActivity != null)
+					return Util.SequenceActivities(NextActivity, new Wait(randFrames), this);
 				else
```

Comment:
If I'm not mistaken, the `else` branch will never be taken since you're queuing up a `Move` activity just a few lines above.